components/utils.py

```diff
@@ -520,3 +606,282 @@ def untar(source, destination='/tmp', strip=1):
     ctx.logger.debug('Extracting {0} to {1}...'.format(source, destination))
     sudo(['tar', '-xzvf', source, '-C', destination,
           '--strip={0}'.format(strip)])
+
+
+def write_file(content, file_path):
+    tmp_file = tempfile.NamedTemporaryFile(delete=False)
+    mkdir(os.path.dirname(os.path.abspath(file_path)))
+    with open(tmp_file.name, 'w') as f:
+        f.write(json.dumps(content))
+    move(tmp_file.name, file_path)
+
+
+class CtxPropertyFactory(object):
+    PROPERTIES_FILE_NAME = 'properties.json'
+    PROPERTIES_PATH = '/opt/cloudify/node_properties/'
```

Comment:
I don't quite understand why we're separating these according to `node_properties`, etc.. and not according to a service.
I think it would make much more sense to create something like `/opt/cloudify/amqpinflux/node_properties`, `/opt/cloudify/amqpinflux/resources`, etc.. for each service which would make handling each service easier. When we move to a distributed manager, the handling of each upgrade mechanism would have to be on the node level.